api/dao/containerutil.py

```diff
@@ -31,14 +36,55 @@ def add_id_to_subject(subject, pid):
         subject['_id'] = bson.ObjectId()
     return subject
 
+def inflate_job_info(analysis):
+    """
+    Inflate job from id ref in analysis
+
+    Lookup job via id stored on analysis
+    Lookup input filerefs and inflate into files array with 'input': True
+    If job is in failed state, look for most recent job referencing this analysis
+    Update analysis if new job is found
+    """
+    if analysis.get('job') is None:
+        return analysis
+    job = config.db.jobs.find_one({'_id': analysis['job']})
+    if job is None:
+        raise Exception('No job with id {} found.'.format(analysis['job']))
```

Comment:
We could also set `job` to `None` here and let the consumer deal with an improper Analysis object. This is called on all analysis for an object so we might not want to bring the whole thing down for one failure. 

If we run into this error we probably have much larger problems, though.